File: lib/enums/display_type.dart
// Display type
/// Defines the display type for permission request dialogs on iOS
///
/// This enum specifies how the permission request interface should be presented
/// to the user when requesting permissions through the Flutter Permission Kit on iOS.
/// The display types are designed to work seamlessly with iOS design guidelines
/// and user experience patterns.
///
/// Available display types:
/// - [alert]: Shows a system-style alert dialog (native iOS alert style)
/// - [modal]: Shows a full-screen modal presentation (iOS modal presentation)
enum DisplayType {
  /// Display permission request as a system alert dialog
  ///
  /// This option presents the permission request in a compact alert dialog
  /// that appears over the current screen content, following iOS Human Interface
  /// Guidelines for alert presentations. It's ideal for simple permission requests
  /// that don't require extensive explanation and maintains the native iOS feel.
  alert,

  /// Display permission request as a full-screen modal
  ///
  /// This option presents the permission request in a full-screen modal
  /// that takes over the entire screen using iOS modal presentation style.
  /// It's ideal for complex permission requests that require detailed explanations
  /// or custom UI elements while maintaining iOS design patterns.
  modal,
}

File: ios/Classes/CorePermissionKit/DisplayType.swift
//
//  DisplayType.swift
//  flutter_permission_kit
//
//  Created by MaoJiu on 2025/5/25.
//

import Foundation

/// Enumeration defining how permission request dialogs should be displayed
///
/// This enum specifies the visual presentation style for permission request
/// dialogs in the iOS implementation of Flutter Permission Kit. Different
/// display types provide different user experiences and visual impacts.
///
/// The display type affects:
/// - Modal presentation style
/// - Animation transitions
/// - User interaction patterns
/// - Visual hierarchy and prominence
///
/// Example usage:
/// ```swift
/// let displayType = DisplayType.alert
/// // or
/// let displayType = DisplayType(from: "modal")
/// ```
enum DisplayType: String {
    /// Alert-style presentation (UIAlertController)
    ///
    /// Presents the permission request as a native iOS alert dialog.
    /// This style provides:
    /// - Native iOS look and feel
    /// - System-standard animations
    /// - Automatic positioning and sizing
    /// - Built-in accessibility support
    /// - Less intrusive user experience
    ///
    /// Best for: Simple permission requests that don't require extensive
    /// explanation or when maintaining native iOS design consistency is important.
    case alert
    
    /// Modal-style presentation (Full-screen or sheet)
    ///
    /// Presents the permission request as a custom modal view controller.
    /// This style provides:
    /// - More space for detailed explanations
    /// - Custom UI design flexibility
    /// - Rich visual content support
    /// - Enhanced branding opportunities
    /// - More prominent user attention
    ///
    /// Best for: Complex permission requests that require detailed explanations,
    /// multiple permissions, or when custom branding is important.
    case modal
    
    /// Initializes a DisplayType from a string value
    ///
    /// This convenience initializer allows creation of DisplayType instances
    /// from string representations, which is particularly useful when parsing
    /// configuration data received from the Flutter layer.
    ///
    /// - Parameter string: The string representation of the display type
    ///   Expected values: "alert" or "modal"
    ///
    /// - Returns: A DisplayType instance if the string matches a valid case,
    ///   or nil if the string doesn't correspond to any known display type
    ///
    /// Example usage:
    /// ```swift
    /// let alertType = DisplayType(from: "alert")     // Returns .alert
    /// let modalType = DisplayType(from: "modal")     // Returns .modal
    /// let invalidType = DisplayType(from: "invalid") // Returns nil
    /// ```
    init?(from string: String) {
        self.init(rawValue: string)
    }
}
